src/main/java/com/bandwidth/sqs/client/SqsClient.java

```diff
@@ -0,0 +1,79 @@
+package com.bandwidth.sqs.client;
+
+import com.amazonaws.regions.Regions;
+import com.bandwidth.sqs.queue.ImmutableSqsQueueClientConfig;
+import com.bandwidth.sqs.queue.SqsQueue;
+import com.bandwidth.sqs.queue.SqsQueueClientConfig;
+import com.bandwidth.sqs.queue.SqsQueueConfig;
+
+import java.text.MessageFormat;
+
+import io.reactivex.Single;
+
+public interface SqsClient {
+
+    /**
+     * @param queueName    Name of the SQS queue. This queue must already exist, it will not be created.
+     * @param region       Region this queue exists in
+     * @param clientConfig Configuration values for the queue client
+     * @return an SqsQueue
+     */
+    Single<SqsQueue<String>> getQueueFromName(String queueName, Regions region, SqsQueueClientConfig clientConfig);
+
+    /**
+     * Gets an SqsQueue object from a queueUrl. Note this does not do any HTTP requests and returns instantly.
+     *
+     * @param queueUrl     Full url of the SQS queue
+     * @param clientConfig Configuration values for the queue client
+     * @return an SqsQueue
+     */
+    SqsQueue<String> getQueueFromUrl(String queueUrl, SqsQueueClientConfig clientConfig);
+
+    /**
+     * Asserts that an SQS queue exists with specific attributes. The queue is created if it does not exist,
+     * and any existing queue is modified if the attributes don't match.
+     *
+     * @param queueConfig  Configuration of the SQS queue
+     * @param clientConfig Configuration of the SQS client
+     * @return an SqsQueue
+     */
+    Single<SqsQueue<String>> assertQueue(SqsQueueConfig queueConfig, SqsQueueClientConfig clientConfig);
```

Comment:
the name `assert` could be construed as implying that this just checks the attributes, and throws an exception if they don't match. `PUT`, as used in HTTP, implies the kind of idempotence that this exhibits, but then actually sounds like you're making some kind of local map.